pkgs/tools/security/ghidra/default.nix

```
@@ -13,58 +21,91 @@
     categories = "Development;";
   };
 
+  # See $sourceRoot/gradle/support/fetchDependencies.gradle
+  flatFileDependencies = {
+    axmlprinter2 = fetchurl {
+      url = "https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/android4me/AXMLPrinter2.jar";
+      sha256 = "0vcxylpx1kqmh6a5km2cix2j3dc1gbns60njr3g6vbxbns707v80";
+    };
+    hfsexplorer = fetchzip {
+      url = "https://sourceforge.net/projects/catacombae/files/HFSExplorer/0.21/hfsexplorer-0_21-bin.zip";
+      sha256 = "08cq6ng16xb5v7mcgv02sljxqwjpd5lvqshm7fqmj7d8spzsasv7";
+      stripRoot = false;
+    };
+    yajsw = fetchurl {
+      url = "https://sourceforge.net/projects/yajsw/files/yajsw/yajsw-stable-12.12/yajsw-stable-12.12.zip";
+      sha256 = "08bkgb55fvvc63pxg0apqjsap2kmwmznv82g5jcikfrsx6qzr60k";
+    };
+  };
 
-in stdenv.mkDerivation rec {
-
+in buildGradle rec {
   pname = "ghidra";
   version = "9.2";
-  versiondate = "20201113";
 
-  src = fetchzip {
-    url = "https://www.ghidra-sre.org/ghidra_${version}_PUBLIC_${versiondate}.zip";
-    sha256 = "0lcvmbq04qkdsf0bz509frgw79bhyxyixkqg1k712p3576ng3nby";
+  src = fetchFromGitHub {
+    owner = "NationalSecurityAgency";
+    repo = pname;
+    rev = "Ghidra_${version}_build";
+    sha256 = "16k2v1ldxsg0hsjns8vd6nbifcib13446pxi86hwdwnxni03p0iw";
   };
 
-  nativeBuildInputs = [
-    makeWrapper
-    autoPatchelfHook
-    icoutils
-  ];
+  envSpec = ./gradle-env.json;
+  nativeBuildInputs = [ bison flex icoutils libarchive makeWrapper ];
+  buildJdk = jdk;
+  gradleFlags = [ "buildGhidra" ];
+
+  # See $sourceRoot/gradle/support/fetchDependencies.gradle
+  postUnpack = with flatFileDependencies; ''
+    mkdir -p $sourceRoot/flatRepo
+    ln -s ${dex2jar}/lib/dex2jar/lib/dex-*.jar $sourceRoot/flatRepo/
+    ln -s ${axmlprinter2} $sourceRoot/flatRepo/AXMLPrinter2.jar
+    ln -s ${hfsexplorer}/lib/{csframework.jar,hfsx_dmglib.jar,hfsx.jar,iharder-base64.jar} $sourceRoot/flatRepo/
+
+    mkdir -p $sourceRoot/Ghidra/Features/GhidraServer/build/
+    ln -s ${yajsw} $sourceRoot/Ghidra/Features/GhidraServer/build/$(stripHash ${yajsw})
+  '';
+
+  postPatch = ''
+    substituteInPlace Ghidra/application.properties \
+      --replace "application.release.name=DEV" "application.release.name=PUBLIC"
 
-  buildInputs = [
-    stdenv.cc.cc.lib
-    pam
-  ];
+    substituteInPlace Ghidra/RuntimeScripts/Common/support/launch.properties \
+      --replace "JAVA_HOME_OVERRIDE=" "JAVA_HOME_OVERRIDE=${jdk.home}"
+  '';
 
-  dontStrip = true;
+  extraInit = ''
+    gradle.projectsLoaded {
+      allprojects {
+        repositories {
+          flatDir name: "flat", dirs:["$rootProject.projectDir/flatRepo"]
+        }
+      }
+    }
+  '';
 
   installPhase = ''
-    mkdir -p "${pkg_path}"
-    mkdir -p "${pkg_path}" "$out/share/applications"
-    cp -a * "${pkg_path}"
+    mkdir -p $out/{bin,share/applications,share/ghidra}
```

```suggestion
    mkdir -p $out/bin $out/share/applications $out/share/ghidra
```

This is easier to read while only marginally longer.